src/app/shared/table-of-contents/table-of-contents.ts

```diff
@@ -0,0 +1,121 @@
+import {
+  Component, Input, AfterViewInit, Inject, ElementRef, OnInit
+} from '@angular/core';
+import {DOCUMENT} from '@angular/platform-browser';
+import {Router, ActivatedRoute, Event, NavigationEnd} from '@angular/router';
+import {Observable} from 'rxjs/Observable';
+import {Subscription} from 'rxjs/Subscription';
+import 'rxjs/add/observable/fromEvent';
+import 'rxjs/add/operator/debounceTime';
+
+@Component({
+  selector: 'table-of-contents',
+  styleUrls: ['./table-of-contents.scss'],
+  template: `
+    <div *ngIf="links?.length" class="toc-container">
+      <h2>Contents</h2>
+      <ul>
```

Comment:
I feel so strange saying this, but I believe the MDN example is wrong and the w3schools is right. [The WAI-ARIA spec specifically says that the list role is for _non-interactive_ content](https://www.w3.org/TR/wai-aria/roles#list).